diff -Naur telepathy-qt-0.9.3/cmake/modules/TpQtMacros.cmake telepathy-qt-0.9.3-tests-install/cmake/modules/TpQtMacros.cmake
--- telepathy-qt-0.9.3/cmake/modules/TpQtMacros.cmake	2012-07-13 17:40:23.000000000 +0000
+++ telepathy-qt-0.9.3-tests-install/cmake/modules/TpQtMacros.cmake	2013-04-12 14:12:41.159804156 +0000
@@ -419,6 +419,7 @@
     set(with_session_bus ${CMAKE_CURRENT_BINARY_DIR}/runDbusTest.sh)
     add_test(${_fancyName} ${SH} ${with_session_bus} ${CMAKE_CURRENT_BINARY_DIR}/test-${_name})
     list(APPEND _telepathy_qt_test_cases test-${_name})
+    list(APPEND _telepathy_qt_test_cases_dbus test-${_name})
 
     # Valgrind and Callgrind targets
     _tpqt_add_check_targets(${_fancyName} ${_name} ${with_session_bus} ${CMAKE_CURRENT_BINARY_DIR}/test-${_name})
diff -Naur telepathy-qt-0.9.3/tests/CMakeLists.txt telepathy-qt-0.9.3-tests-install/tests/CMakeLists.txt
--- telepathy-qt-0.9.3/tests/CMakeLists.txt	2012-07-13 17:40:23.000000000 +0000
+++ telepathy-qt-0.9.3-tests-install/tests/CMakeLists.txt	2013-04-12 20:34:51.013462217 +0000
@@ -42,6 +42,24 @@
 tpqt_add_generic_unit_test(RCCSpec rccspec)
 tpqt_add_generic_unit_test(FileTransferChannelCreationProperties file-transfer-channel-creation-properties)
 
+# Add Nemo install targets
+set(NEMO_TESTS_DIR "/opt/tests/telepathy-qt${QT_VERSION_MAJOR}")
+set(NEMO_TESTS_DIR_BIN "${NEMO_TESTS_DIR}/bin")
+set(NEMO_TESTS_DIR_LIB "${NEMO_TESTS_DIR}/lib")
+set(NEMO_TESTS_DIR_DATA "${NEMO_TESTS_DIR}/data")
+set(NEMO_TESTS_DIR_DBUS "${NEMO_TESTS_DIR}/dbus-1")
+configure_file(mktests.sh.in ${CMAKE_CURRENT_SOURCE_DIR}/mktests.sh)
+configure_file(runDbusTest.sh.in ${CMAKE_CURRENT_SOURCE_DIR}/runDbusTest.sh)
+configure_file(runTest.sh.in ${CMAKE_CURRENT_SOURCE_DIR}/runTest.sh)
+file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/unit-tests.list "${_telepathy_qt_test_cases}")
+INSTALL(TARGETS ${_telepathy_qt_test_cases} RUNTIME DESTINATION ${NEMO_TESTS_DIR_BIN})
+INSTALL(FILES ${CMAKE_CURRENT_SOURCE_DIR}/runTest.sh ${CMAKE_CURRENT_SOURCE_DIR}/runDbusTest.sh DESTINATION ${NEMO_TESTS_DIR_BIN} PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
+INSTALL(FILES ${CMAKE_CURRENT_SOURCE_DIR}/INSIGNIFICANT ${CMAKE_CURRENT_SOURCE_DIR}/tests.xml DESTINATION ${NEMO_TESTS_DIR})
+INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/unit-tests.list DESTINATION ${NEMO_TESTS_DIR})
+INSTALL(FILES ${CMAKE_CURRENT_SOURCE_DIR}/test-key-file-format-error.ini ${CMAKE_CURRENT_SOURCE_DIR}/test-key-file.ini DESTINATION ${NEMO_TESTS_DIR_DATA}/tests)
+INSTALL(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/telepathy DESTINATION ${NEMO_TESTS_DIR_DATA}/tests)
+
 add_subdirectory(dbus-1)
 add_subdirectory(dbus)
 add_subdirectory(lib)
+
diff -Naur telepathy-qt-0.9.3/tests/dbus/CMakeLists.txt telepathy-qt-0.9.3-tests-install/tests/dbus/CMakeLists.txt
--- telepathy-qt-0.9.3/tests/dbus/CMakeLists.txt	2012-07-13 17:40:23.000000000 +0000
+++ telepathy-qt-0.9.3-tests-install/tests/dbus/CMakeLists.txt	2013-04-12 20:35:46.637635377 +0000
@@ -100,3 +100,9 @@
 add_custom_target(check ctest --output-on-failure -O test.log
                   WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/tests)
 add_dependencies(check check-local ${_telepathy_qt_test_cases})
+
+# Add Nemo install target for tests
+INSTALL(TARGETS ${_telepathy_qt_test_cases_dbus} RUNTIME DESTINATION ${NEMO_TESTS_DIR_BIN})
+file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/dbus-tests.list "${_telepathy_qt_test_cases_dbus}")
+INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/dbus-tests.list DESTINATION ${NEMO_TESTS_DIR})
+INSTALL(FILES ${CMAKE_SOURCE_DIR}/tools/with-session-bus.sh DESTINATION ${NEMO_TESTS_DIR_BIN} PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
diff -Naur telepathy-qt-0.9.3/tests/dbus-1/CMakeLists.txt telepathy-qt-0.9.3-tests-install/tests/dbus-1/CMakeLists.txt
--- telepathy-qt-0.9.3/tests/dbus-1/CMakeLists.txt	2012-07-13 17:40:23.000000000 +0000
+++ telepathy-qt-0.9.3-tests-install/tests/dbus-1/CMakeLists.txt	2013-04-12 19:10:43.620323663 +0000
@@ -2,3 +2,7 @@
 configure_file(session.conf.in ${CMAKE_CURRENT_BINARY_DIR}/session.conf)
 
 add_subdirectory(services)
+
+# Add Nemo install target
+INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/session.conf DESTINATION ${NEMO_TESTS_DIR_DBUS})
+
diff -Naur telepathy-qt-0.9.3/tests/dbus-1/services/CMakeLists.txt telepathy-qt-0.9.3-tests-install/tests/dbus-1/services/CMakeLists.txt
--- telepathy-qt-0.9.3/tests/dbus-1/services/CMakeLists.txt	2012-07-13 17:40:23.000000000 +0000
+++ telepathy-qt-0.9.3-tests-install/tests/dbus-1/services/CMakeLists.txt	2013-04-12 19:09:13.180992764 +0000
@@ -2,3 +2,7 @@
 set(abs_top_srcdir ${CMAKE_SOURCE_DIR})
 configure_file(account-manager.service.in ${CMAKE_CURRENT_BINARY_DIR}/account-manager.service)
 configure_file(spurious.service.in ${CMAKE_CURRENT_BINARY_DIR}/spurious.service)
+
+# Add Nemo install target
+INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/spurious.service ${CMAKE_CURRENT_BINARY_DIR}/account-manager.service DESTINATION ${NEMO_TESTS_DIR_DBUS}/services/)
+
diff -Naur telepathy-qt-0.9.3/tests/dbus-1/services/account-manager.service.in telepathy-qt-0.9.3-tests-install/tests/dbus-1/services/account-manager.service.in
--- telepathy-qt-0.9.3/tests/dbus-1/services/account-manager.service.in	2013-04-12 17:55:05.490364840 +0000
+++ telepathy-qt-0.9.3-tests-install/tests/dbus-1/services/account-manager.service.in	2013-04-12 18:49:27.852722717 +0000
@@ -1,3 +1,3 @@
 [D-BUS Service]
 Name=org.freedesktop.Telepathy.AccountManager
-Exec=@TEST_PYTHON@ @abs_top_srcdir@/tests/lib/python/account-manager.py
+Exec=@TEST_PYTHON@ @NEMO_TESTS_DIR_BIN@/account-manager.py
diff -Naur telepathy-qt-0.9.3/tests/dbus-1/session.conf.in telepathy-qt-0.9.3-tests-install/tests/dbus-1/session.conf.in
--- telepathy-qt-0.9.3/tests/dbus-1/session.conf.in	2012-07-13 17:40:23.000000000 +0000
+++ telepathy-qt-0.9.3-tests-install/tests/dbus-1/session.conf.in	2013-04-12 18:49:12.452663093 +0000
@@ -12,7 +12,7 @@
 
   <listen>unix:tmpdir=/tmp</listen>
 
-  <servicedir>@abs_top_builddir@/tests/dbus-1/services/</servicedir>
+  <servicedir>@NEMO_TESTS_DIR_DBUS@/services/</servicedir>
 
   <policy context="default">
     <!-- Allow everything to be sent -->
diff -Naur telepathy-qt-0.9.3/tests/lib/CMakeLists.txt telepathy-qt-0.9.3-tests-install/tests/lib/CMakeLists.txt
--- telepathy-qt-0.9.3/tests/lib/CMakeLists.txt	2012-07-13 17:40:23.000000000 +0000
+++ telepathy-qt-0.9.3-tests-install/tests/lib/CMakeLists.txt	2013-04-12 20:37:12.523909257 +0000
@@ -28,3 +28,7 @@
     add_subdirectory(glib)
     add_subdirectory(glib-helpers)
 endif(ENABLE_TP_GLIB_TESTS)
+
+# Add Nemo install target
+INSTALL(TARGETS tp-qt-tests LIBRARY DESTINATION ${NEMO_TESTS_DIR_LIB})
+INSTALL(FILES ${CMAKE_CURRENT_SOURCE_DIR}/python/account-manager.py DESTINATION ${NEMO_TESTS_DIR_BIN} PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
diff -Naur telepathy-qt-0.9.3/tests/lib/glib/CMakeLists.txt telepathy-qt-0.9.3-tests-install/tests/lib/glib/CMakeLists.txt
--- telepathy-qt-0.9.3/tests/lib/glib/CMakeLists.txt	2012-07-13 17:40:23.000000000 +0000
+++ telepathy-qt-0.9.3-tests-install/tests/lib/glib/CMakeLists.txt	2013-04-12 17:48:39.140353221 +0000
@@ -71,3 +71,7 @@
         tp-glib-tests-future-extensions)
 
 endif(ENABLE_TP_GLIB_TESTS)
+
+# Add Nemo install target
+INSTALL(TARGETS tp-glib-tests LIBRARY DESTINATION ${NEMO_TESTS_DIR_LIB})
+
diff -Naur telepathy-qt-0.9.3/tests/lib/glib-helpers/CMakeLists.txt telepathy-qt-0.9.3-tests-install/tests/lib/glib-helpers/CMakeLists.txt
--- telepathy-qt-0.9.3/tests/lib/glib-helpers/CMakeLists.txt	2012-07-13 17:40:23.000000000 +0000
+++ telepathy-qt-0.9.3-tests-install/tests/lib/glib-helpers/CMakeLists.txt	2013-04-12 17:49:01.582392678 +0000
@@ -20,3 +20,6 @@
         ${QT_QTDBUS_LIBRARY}
         telepathy-qt${QT_VERSION_MAJOR})
 endif(ENABLE_TP_GLIB_TESTS)
+
+# Add Nemo install target
+INSTALL(TARGETS tp-qt-tests-glib-helpers LIBRARY DESTINATION ${NEMO_TESTS_DIR_LIB})
